test/unittest/tests/test_string_match.cpp

```diff
@@ -0,0 +1,121 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include <catch2/catch.hpp>
+#include "wt_internal.h"
+
+TEST_CASE("Test string matching macro: WT_STRING_MATCH", "[STRING_MATCH]")
+{
+    WT_CONFIG_ITEM green, empty;
+    int i;
+    const char *empty_str, *g_str, *greed_str, *green_str, *greener_str;
+    const char *null_str, *something_str;
+
+    /* The "right side" of the match */
+    green.str = "green";
+    green.len = 5;
+    empty.str = "";
+    empty.len = 0;
+
+    /* The "left side" of the match */
+    empty_str = "";
+    g_str = "g";
+    greed_str = "greed";
+    green_str = "green";
+    greener_str = "greener";
+    null_str = NULL;
+    something_str = "greener";
+
+    /*
+     * Thwart smart compilers: if they know this variable is null, they'll complain about its use
+     * with standard string functions, which may be used by the macros we are testing.
+     */
+    if (getenv("__never__") != NULL)
+        null_str = "null";
+
+    SECTION("Check for literal string matching")
+    {
+        CHECK(null_str == NULL);
+
+        /* We run the test twice, once with the config string null terminated, one without. */
+        for (i = 0; i < 2; i++) {
+            CHECK(WT_STRING_LIT_MATCH("g", green.str, green.len) == 0);
```

Comment:
I think a comment here because, while WT code is often `if (WT_STRING_MATCH(...)) <do something>` is clear, these lines are confusing because `strcmp == 0` means a match and here it means doesn't match. So all the explicit checks make me really think about what is going on. Maybe even define a `MATCHES` or `DOESNT_MATCH` but that could be overkill.

Reply:
Would these be clearer as `== false` and `== true`?

Reply:
Yes.

Reply:
Changed to false/true.